client/cmd/dexc-desktop/syncserver.go

```diff
@@ -0,0 +1,172 @@
+		return
+	}
+	if addr == "" {
+		log.Errorf("No sync address found for kill signal: %v", err)
+		return
+	}
+	resp, err := http.Get("http://" + addr + "/kill")
+	if err != nil {
+		log.Errorf("Error sending kill signal to sync server: %v", err)
+		return
+	}
+	if resp == nil {
+		log.Errorf("nil kill response")
+		return
+	}
+	if resp.StatusCode == http.StatusOK {
+		log.Errorf("Unexpected response code from kill signal send: %d (%s)", resp.StatusCode, http.StatusText(resp.StatusCode))
+	}
+}
+
+// synchronize attempts to determine the state of any other dexc-desktop
+// instances that are running. If none are found, startServer will be true.
+// If another instance is found, close will be true, and this instance of
+// dexc-desktop should exit immediately.
+func synchronize(syncDir string) (startServer, close bool, err error) {
+	addr, err := getSyncAddress(syncDir)
+	if err != nil {
+		return false, false, err
+	}
+	if addr == "" {
+		// Just start the server then.
+		return true, false, nil
+	}
+	resp, err := http.Get("http://" + addr)
+	if err == nil && resp.StatusCode == http.StatusOK {
+		// Other instance will open the window.
+		return false, true, nil
+	}
+	return true, false, nil
+}
+
+// runServer runs an instance of the sync server. Received commands are
+// communicated via unbuffered channels. Blocking channels are ignored.
+func runServer(ctx context.Context, syncDir string, openC chan<- struct{}, killC chan<- os.Signal) {
+	addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
+	if err != nil {
+		log.Errorf("ResolveTCPAddr error: %v", err)
+		return
+	}
+
+	l, err := net.ListenTCP("tcp", addr)
+	if err != nil {
+		log.Errorf("ListenTCP error: %v", err)
+		return
+	}
+
+	f, err := os.Create(filepath.Join(syncDir, syncFilename))
+	if err != nil {
+		log.Errorf("Failed to start the sync server: %v", err)
+		return
+	}
+	// Write the address to the syncfile.
+	if _, err := f.Write([]byte(l.Addr().String())); err != nil {
+		log.Errorf("Error writing syncfile: %v", err)
+	}
+	if err := f.Close(); err != nil {
+		log.Errorf("Error closing syncfile: %v", err)
+	}
+
+	srv := &http.Server{
+		Addr:    l.Addr().String(),
+		Handler: &syncServer{openC, killC},
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		if err := srv.Serve(l); !errors.Is(err, http.ErrServerClosed) {
+			log.Errorf("listen: %s\n", err)
+		}
+		log.Infof("Sync server off")
+	}()
+
+	<-ctx.Done()
+	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
+	defer cancel()
+	if err := srv.Shutdown(ctx); err != nil {
+		log.Errorf("http.Server Shutdown errored: %v", err)
+	}
+	wg.Wait()
+}
+
+type syncServer struct {
+	openC chan<- struct{}
+	killC chan<- os.Signal
+}
+
+func (s *syncServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusOK)
+	switch r.URL.Path {
+	case "/":
+		select {
+		case s.openC <- struct{}{}:
+			log.Info("Window reopened")
```

Comment:
**Nit:** technically "Window reopened" message here should be something like "Received/handling window reopen request" since sending to `s.openC` does not guarantee successfully opening window (which happens in parallel go-routine, and we don't get any confirmations about the result here),

same applies to "Window opened" msg in another place in this PR